### ISSUE TITLE
Custom object serialization while using the HttpProxy

### ISSUE THREAD

**Issue by GreatBarrier86** [MarimerLLC/cslaforum#738]
**Question**
We're trying to use the standard `HttpProxy` object provided by the CSLA library when saving object data. By default, when the `MobileFormatter` serializes the `FieldManager`, it brings in all registered properties. In our case, this presents a problem because there are times we don't want to bring the whole kitchen sink over. For instance, when we save rotated photos, we don't want to resend the photo's `byte[]`, but instead send an integer that represents how many times the photo should be rotated on the server side.

According to the link [here](https://github.com/MarimerLLC/cslaforum/issues/624), it's implied that i should override `OnGetState` and return only the properties i need, but like that link also says, there is no way to set the `FieldManager` instance on `BusinessBase` to a subclass with that override, so what is the recommended solution for how to exclude certain properties? In addition, there is no way to replace `MobileFormatter`, so i can't attack it from that direction either.

Simply put, how would I modify what gets added from this method?
![image](https://user-images.githubusercontent.com/29482393/56508607-b864fc80-64f2-11e9-8fcb-c3dca541ce83.png)

**Version and Platform**
CSLA version: 4.9
OS: Windows
Platform: Xamarin

**Comment by jonnybee** [comment]
Assuming that you contol the code that call the Rotate/Save function you could simply set that value to null and set the number of times to rotate the image on the server side before you call DataPortal.Update. 

Another option would be to create a separate CommandObject that does the RotateImage function.

**Comment by rockfordlhotka** [comment]
If you want a non-serialized property value (or sometimes serialized property value) you need to use a private backing field, not a managed backing field.

A private backing field _requires_ that you implement `OnGetState`/`OnSetState` to serialize/deserialize the backing field, which should allow you to choose whether to serialize it or not.

Keep in mind that if you are serializing child objects like a `MobileList` then there are get/set methods you should override for child object serialization. That happens at a different point in the serialization process than per-instance field serialization and you'll run into trouble if you don't let the serializer do its job.

**Comment by Wandar2000** [comment]
I would vote up to have the fieldmanager to have as a factory into BB to be able to override it in stead of to have to alter CSLA to obtain the goals to have in issue 624.

**Comment by GreatBarrier86** [comment]
624? That's this one: https://github.com/MarimerLLC/csla/issues/624

Did you mean something else?

**Comment by GreatBarrier86** [comment]
@rockfordlhotka, i must not understand. Properties are serialized when the MobileFormatter serializes the _fieldManager, not the object itself, correct? Since FieldManager can't be changed, i can't override OnGetState/OnSetState to exclude certain properties in certain situations.

What am i misunderstanding?

**Comment by rockfordlhotka** [comment]
I just answered a similar question in another thread, though I can't find it just at the moment.

If you have a field that you want to serialize sometimes and not others then you must use a private backing field and override the OnGet/SetState methods to serialize the value (or not) based on your rules. Properties that use a managed backing field are always serialized.

Also, if your property/field refers to a complex object then you need to override the get/set _child_ methods, otherwise MobileFormatter won't be able to do the right things to manage the child object graph.

**Comment by GreatBarrier86** [comment]
Ah, OK. Now i understand. That's not particularly ideal for our situation.

**One way we considered was this**
1. Override `BusinessObject.SaveAsync`
2. Store off the byte[] property's value to a local variable
3. Call `LoadProperty(ByteArrayProperty, null)`. By using `LoadProperty`, we don't dirty the object, so the web service won't commit that `NULL` to the DB.
4. Do `base.SaveAsync`
5. Call `returnedObject.LoadProperty(ByteArrayProperty, savedLocalVariable)`.

Do you see any reason why this might be a problem?

**Comment by GreatBarrier86** [comment]
Understood. That clarifies everything! thank you.
